lib/agents/index.ts

```diff
@@ -0,0 +1,161 @@
+import * as cdk from "aws-cdk-lib";
+import { Construct } from "constructs";
+import * as lambda from "aws-cdk-lib/aws-lambda";
+import * as iam from "aws-cdk-lib/aws-iam";
+import * as path from "path";
+import * as geo from "aws-cdk-lib/aws-location";
+import { bedrock } from "@cdklabs/generative-ai-cdk-constructs";
+import { Bucket } from "aws-cdk-lib/aws-s3";
+import { BucketDeployment, Source } from "aws-cdk-lib/aws-s3-deployment";
+import { VectorIndex } from "@cdklabs/generative-ai-cdk-constructs/lib/cdk-lib/opensearch-vectorindex";
+import { VectorCollection } from "@cdklabs/generative-ai-cdk-constructs/lib/cdk-lib/opensearchserverless";
+import { NagSuppressions } from "cdk-nag";
+
+export class BedrockWeatherAgent extends Construct {
+  constructor(scope: Construct, id: string) {
+    super(scope, id);
+
+    const powertools = lambda.LayerVersion.fromLayerVersionArn(
+      this,
+      "powertools",
+      `arn:aws:lambda:${
+        cdk.Stack.of(this).region
+      }:017000801446:layer:AWSLambdaPowertoolsPythonV2:58`
+    );
+
+    const placeIndex = new geo.CfnPlaceIndex(this, "place-index", {
```

Comment:
nit: other classes use this case for construct ids 
```suggestion
    const placeIndex = new geo.CfnPlaceIndex(this, "PlaceIndex", {
```